agents-for-bedrock/use-case-examples/text-2-sql-agent-cdk-enhanced/app.py

```diff
@@ -0,0 +1,225 @@
+                "glue:BatchGetPartition", "glue:GetPartition", "glue:GetPartitions",
+                "glue:BatchCreatePartition", "glue:CreatePartition", "glue:DeletePartition",
+                "glue:UpdatePartition", "glue:BatchDeletePartition"
+            ],
+            resources=[f"arn:aws:glue:{region}:{account_id}:catalog",
+                       f"arn:aws:glue:{region}:{account_id}:database/{glue_database_name}",
+                       f"arn:aws:glue:{region}:{account_id}:table/{glue_database_name}/*"]
+        ))
+        action_group_function.role.add_to_policy(iam.PolicyStatement(
+            actions=["s3:PutObject", "s3:GetObject", "s3:ListBucket","s3:CreateBucket", "s3:GetBucketLocation"],
+            resources=[f"{schema_bucket.bucket_arn}", f"{schema_bucket.bucket_arn}/*"]
+        ))
+        action_group_function.role.add_to_policy(iam.PolicyStatement(
+            actions=[
+                "athena:StartQueryExecution", "athena:GetQueryExecution",
+                "athena:GetQueryResults", "athena:StopQueryExecution",
+                "athena:GetWorkGroup"
+            ],
+            resources=[f"arn:aws:athena:{region}:{account_id}:workgroup/primary"]
+        ))
+
+        
+        
+        data_folder = f'./Data/{data_folder_name}'
+
+        data_context = analyze_csv_files(data_folder)
+        
+        instruction_text = generate_instruction(data_context, glue_database_name)
+        # question = f"""
+        #     Please craft a comprehensive, single-paragraph instruction (up to 1200 words) for the Bedrock agent. Incorporate all the contextual details provided below to develop this instruction text. The final submission should consist solely of this detailed instruction.\n
+        #     {instruction_text}.
+        #     """ 
+            
+        
+        question = f"""
+            Craft a comprehensive and cohesive paragraph instruction for the Bedrock agent, ensuring the instruction text includes all 7 contextual details and examples provided. The instruction should be detailed, precise with a maximum length of 4000 characters. Clearly outline the agent's tasks and how it should interact with users, incorporating the provided contextual details and examples with minimal changes. Avoid any introductory phrases such as "Here is your...".
+
+            <Contextual details and examples>
+            {instruction_text}
+            """
+   
+                 
+        agent_instruction=invoke_claude_3_with_text(question)    
+        print(agent_instruction)
+        
+        api_schema = ApiSchema.from_asset("./text_to_sql_openapi_schema.json")
+        agent = Agent(
+            self,
+            "MyAgent",
+            foundation_model=foundation_model,
+            instruction=agent_instruction,
+            description="Agent for performing sql queries.",
+            should_prepare_agent=True
+          
+        )
+        # agent.add_alias(self, 'ProdAlias', 
```

Comment:
there is issue with this and opened already an issue and it will be fixed in future (for now, we are good without it)
https://github.com/awslabs/generative-ai-cdk-constructs/issues/428#:~:text=thank%20you%20for%20reporting%20this%20issue%2C%20I%20was%20able%20to%20reproduce%2C%20will%20update%20this%20ticket%20when%20a%20fix%20is%20available